kafka/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/AzureAuthenticationCallbackHandler.scala

```diff
@@ -0,0 +1,71 @@
+/**
+  * Copyright (c) 2013-present Snowplow Analytics Ltd.
+  * All rights reserved.
+  *
+  * This program is licensed to you under the Snowplow Community License Version 1.0,
+  * and you may not use this file except in compliance with the Snowplow Community License Version 1.0.
+  * You may obtain a copy of the Snowplow Community License Version 1.0 at https://docs.snowplow.io/community-license-1.0
+  */
+package com.snowplowanalytics.snowplow.collectors.scalastream
+package sinks
+
+import java.net.URI
+import java.{lang, util}
+
+import javax.security.auth.callback.Callback
+import javax.security.auth.callback.UnsupportedCallbackException
+import javax.security.auth.login.AppConfigurationEntry
+
+import org.apache.kafka.clients.producer.ProducerConfig
+import org.apache.kafka.common.security.auth.AuthenticateCallbackHandler
+import org.apache.kafka.common.security.oauthbearer.OAuthBearerToken
+import org.apache.kafka.common.security.oauthbearer.OAuthBearerTokenCallback
+
+import com.microsoft.azure.credentials.MSICredentials
+
+import com.nimbusds.jwt.JWTParser
+
+class AzureAuthenticationCallbackHandler extends AuthenticateCallbackHandler {
+
+  val credentials: MSICredentials = new MSICredentials()
+
+  var sbUri: String = ""
+
+  override def configure(
+    configs: util.Map[String, _],
+    saslMechanism: String,
+    jaasConfigEntries: util.List[AppConfigurationEntry]
+  ): Unit = {
+    val bootstrapServer = configs.get(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG).toString.replaceAll("\\[|\\]", "")
```

Comment:
Out of curiosity, please can you share an example of what this bootstrap server config looks like?  I'm curious about which part of it needs replacing/removing.

[The Kafka documentation](https://docs.confluent.io/platform/current/installation/configuration/producer-configs.html#bootstrap-servers) says bootstrap servers can be a comma separated list of multiple servers.  Should we also split by comma and take the first host?

Reply:
> Out of curiosity, please can you share an example of what this bootstrap server config looks like? I'm curious about which part of it needs replacing/removing.

It looks like `host1:port1,host2:port2`. I don't think it has any `[]` but I've added `replaceAll` part just in case since they had that in [the example](https://github.com/Azure/azure-event-hubs-for-kafka/blob/master/tutorials/oauth/java/managedidentity/consumer/src/main/java/CustomAuthenticateCallbackHandler.java#L41).

> [The Kafka documentation](https://docs.confluent.io/platform/current/installation/configuration/producer-configs.html#bootstrap-servers) says bootstrap servers can be a comma separated list of multiple servers. Should we also split by comma and take the first host?

Yes, it should be done like that. Good catch! I will update it.